Soccar [UI]/src/nl/soccar/ui/fx/drawable/CarUiFx.java

```diff
@@ -54,10 +59,21 @@ public CarUiFx(GameCanvas canvas, Car car, CarPhysics physics, TeamColour colour
     @Override
     public void draw(GraphicsContext context) {
         CarPhysics physics = super.getPhysicsModel();
-        
-        super.getModel().setSteerAction(Keyboard.getSteerAction());
-        super.getModel().setHandbrakeAction(Keyboard.getHandbrakeAction());
-        super.getModel().setThrottleAction(Keyboard.getThrottleAction());
+
+        if (ClientController.getInstance().getCurrentPlayer().getUsername().equals(super.getModel().getPlayer().getUsername())) {
+            SteerAction steerAction = Keyboard.getSteerAction();
+            HandbrakeAction handbrakeAction = Keyboard.getHandbrakeAction();
+            ThrottleAction throttleAction = Keyboard.getThrottleAction();
+
+            super.getModel().setSteerAction(steerAction);
+            super.getModel().setHandbrakeAction(handbrakeAction);
+            super.getModel().setThrottleAction(throttleAction);
+
+            // TODO : Change location of sending this message (Ugly implementation for now)

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1135)